sqlx-core/src/types/mod.rs

```diff
@@ -99,8 +99,8 @@ pub use bstr::{BStr, BString};
 ///
 /// ## Compile-time verification
 ///
-/// With compile-time verification, the use of type overrides is currently required to make
-/// use of any user-defined types.
+/// The use of type overrides is currently required to make use of any user-defined types,
+/// so they cannot be compile-time verified.
```

Comment:
If we're revising this statement, we might as well make it abundantly clear:

```suggestion
/// Type definitions are *not* verified against the database at compile-time.
/// The [`query!()`][sqlx::query!] macros have no implicit knowledge of user-defined types.
///
/// When using custom types in query parameters or output columns with `query!()`, 
/// the use of [type overrides][sqlx::query!#type-overrides-bind-parameters-postgres-only] is required.
```